mathjax3-ts/input/tex/ParseUtil.ts

```diff
@@ -488,6 +488,30 @@ namespace ParseUtil {
     return (font ? {mathvariant: font} : {});
   };
 
+
+  /**
+   * Splits a package option list of the form [x=y,z=1] into an attribute list
+   * of the form {x: y, z: 1}.
+   * @param {string} attrib The attributes of the package.
+   * @param {{[key: string]: number}?} allowed A list of allowed options.
+   * @return {EnvList} The attribute list.
+   */
+  export function splitPackageOptions(attrib: string, allowed: {[key: string]: number} = {}): EnvList {
+    let def: EnvList = {};
+    if (attrib !== '') {
+      const attr = attrib.replace(/ /g, '').split(/,/);
+      for (let i = 0, m = attr.length; i < m; i++) {
+        const keyvalue = attr[i].split(/[:=]/);
+        if (allowed[keyvalue[0]]) {
```

Comment:
This should probably be `allowed.hasOwnProperty(keyvalue[0])`, since otherwise something like `[toString=1]` would be allowed.

Reply:
Outdated.